processor/resourcedetectionprocessor/internal/gcp/gcp.go

```diff
@@ -52,6 +53,18 @@ type detector struct {
 }
 
 func (d *detector) Detect(context.Context) (resource pcommon.Resource, schemaURL string, err error) {
+	if d.detector.CloudPlatform() == gcp.BareMetalSolution {
+		d.rb.SetCloudProvider(conventions.AttributeCloudProviderGCP)
```

Comment:
is bare metal solutions actually on GCP?

Reply:
Yes, Bare Metal Solution is part of Google Cloud Platform